docs/proposals/registry/registry-structure.md

```diff
@@ -0,0 +1,101 @@
+
+**Registry Build:** The process that takes the devfile registry repository, generates the index.json and builds it up into the devfile index container image.
+
+**Registry Bootstrap:** The process that pushes the devfile stacks on the devfile index container to the OCI registry.
+
+## As-is Today
+Currently, the top-level structure of a devfile registry’s repository is unwritten, but it usually consists of a **devfiles** or **stacks** folder, and any associated files specific to that registry
+
+[devfile/registry](https://github.com/devfile/registry):
+<img width="914" alt="Screen Shot 2020-11-30 at 2 12 13 PM" src="https://user-images.githubusercontent.com/6880023/100653219-12a48100-3316-11eb-949c-38073a19acbc.png">
+
+Inside each **stacks** or **devfiles** folder, each folder corresponds to a devfile stack, containing usually just the devfile.yaml and a meta.yaml:
+<img width="1232" alt="Screen Shot 2020-11-30 at 2 10 50 PM" src="https://user-images.githubusercontent.com/6880023/100653095-e2f57900-3315-11eb-8c3f-86e56896ef15.png">
+
+<img width="1229" alt="Screen Shot 2020-11-30 at 2 11 20 PM" src="https://user-images.githubusercontent.com/6880023/100653133-f0aafe80-3315-11eb-8393-dbabfd8ce0d6.png">
+
+As part of the registry build, the index.json is generated based off of the stacks in the repository, and a devfile index image is generated containing the index.json and stacks.
+
+When deploying the OCI registry, the registry bootstrap process parses the index.json to find the devfile.yaml for each stack, and pushes the devfile up to the registry as a single layer. No other stack artifacts are pushed up as part of the layer, or as separate layer.
+
+## Problem
+Our approach of pushing only the devfile.yaml works fine currently because the stacks we’re pushing only have a devfile.yaml in them. However, devfile stacks may contain more than _just_ the devfile, and may also contains resources that are re-used across multiple stacks (such as certain vsx plugins).
+
+We need a defined way of knowing what files in a stack to push up in a, what layers each file in the stack should belong to, and where the stack's files should located. 
+
+## Proposal
+
+To solve the issue listed above, we should:
+1) Formally the expected structure of the Devfile Regisry Repository.
+2) Define the layers that compose a devfile stack on an OCI registry, and what each layer contains.
+
+### Repository Structure
+The structure of the “Devfile Registry Repository” should impose the following requirements:
+1) A top-level folder called `stacks`, which contains folders for each devfile stack.
+2) Each devfile stack folder must contain a `devfile.yaml`. Other files such as vsx plugins, stack logos, etc. can be included as needed.
+3) A Dockerfile to package the stacks and index.json (see https://github.com/devfile/api/blob/master/docs/proposals/registry/devfile-packaging.md)
```

Comment:
I think we need to be careful about the role of Dockerfile on the top level of the stack. In devfile 2.2 where we introduce the outer loop support, a stack may contain a Dockerfile for building the user application. If we put the packaging stack Dockerfile on the top level of the stack folder, we'll have problems in devfile 2.2. We can consider putting the packaging stack Dockerfile to a reserved folder, e.g. `buildResources`, to avoid potential conflicts in the future.

Reply:
So the Dockerfile and build tools mentioned here actually the registry dockerfile and build tools, not for each individual stack. I just wanted to include them since we were talking about the structure of the devfile registry repository. I'll change it to be something like:
```
In addition to the registry's Dockerfile and build tools, the following requirements are imposed on the devfile registry repository:
 1) A top-level folder called `stacks`, which contains folders for each devfile stack.
2) Each devfile stack folder must contain a `devfile.yaml`. Other files such as vsx plugins, stack logos, etc. can be included as needed. 
```

That being said, did we want to include a reserved folder for the stack's build tools as well, or just leave it up to each stack author?

Reply:
Updated to remove the mentions of the Dockerfile and build scripts for the registry, as discussed on today's meeting.

The registry build tools and Dockerfile will be moved to the registry-support repo.